### DISCUSSION TITLE
The problem I have met

### DISCUSSION THREAD

**Discussion by Leason-ly** [guardstrikelab/carla_apollo_bridge#24] [Q&A]
Hello, I've followed your guide to setup. However, there was a problem at the and of the bridge. When I run the 
file called "carla_cyber_bridge/bridge.py", I'm confronted with an error.

I1121 07:44:25.121304  7055 global_data.cc:119] [cyber_py] host ip: 172.17.0.2
[2022-11-21 07:44:25,123] [carla_bridge] [INFO]: The config file path is carla_cyber_bridge/config/settings.yaml.
[2022-11-21 07:44:25,125] [carla_bridge] [INFO]: Trying to connect to 172.17.0.2:2000
[2022-11-21 07:44:55,128] [carla_bridge] [ERROR]: Error: time-out of 30000ms while waiting for the simulator, make sure the simulator is ready and connected to 172.17.0.2:2000
Traceback (most recent call last):
  File "carla_cyber_bridge/bridge.py", line 487, in <module>
    main()
  File "carla_cyber_bridge/bridge.py", line 473, in main
    carla_world = carla_client.get_world()
RuntimeError: time-out of 30000ms while waiting for the simulator, make sure the simulator is ready and connected to 172.17.0.2:2000

I wish you can help me solve the problem, th-x!

**Comment by synkrotron** [comment]
It seems The Carla server did not connected. Please try to connect Carla server with "172.17.0.1"

**Reply by Leason-ly** [reply]
thanks，but i have met another problem. I successfully connected to carla and generated the scene. But there is no response on the dreamview side.
when I open dreamview,i found there is no carla_town01 map.
![1](https://user-images.githubusercontent.com/73697516/203235805-767996ca-b359-41a5-83e3-4cc9218186e6.png)

I checked its source code and found “carla_ Town01” is in the "modules/map/carla_town01" folder, not in the "modules/map/data" folder.
When I run the dreamview module with the "./script/dreamview. sh start_fe" command, I get an error.
![2](https://user-images.githubusercontent.com/73697516/203235764-beb28ed7-dc80-4dd7-bf7a-379f79e8fd8b.png)

So，should I move the files to the ”data“ folder?



![3](https://user-images.githubusercontent.com/73697516/203237294-aef49365-0bed-4845-877e-246d5ecaede2.png)

**Reply by synkrotron** [reply]
Are you using 'guardstrike/apollo_carla_bridge' branch of https://github.com/guardstrikelab/apollo.git ?

**Reply by synkrotron** [reply]
We have a little bit changes of Apollo source code.

**Reply by Leason-ly** [reply]
yep， i followed the guildline,using "
#Using HTTPS
git clone --branch guardstrike/apollo_carla_bridge https://github.com/guardstrikelab/apollo.git
"
The source code I cloned is exactly what I showed earlier.

**Reply by Leason-ly** [reply]
And，when i move the “carla_ Town01” to "modules/map/data" folder, there is still a problem.

![image](https://user-images.githubusercontent.com/73697516/203252122-b4a8035e-dbb2-431d-868f-5b4f88dbfb70.png)

I THINK maybe the version of the Apollo source code you commit is wrong.

I hope you can help me. THANKS A LOT.

BEST WISHES.

**Comment by synkrotron** [comment]
@SEEKERXC @wohu1104 Please look at this question when you are free.

**Comment by wohu1104** [comment]
@Leason-ly    I see "permission denied" in the screenshot, please make sure the permission is correct.

**Reply by Leason-ly** [reply]
I know this，but i‘ve try it with "sudo" command again. And the "permission" is to create a log file to record the results of the running.
Are you sure the version you commit is right?
And as I asked before, should i move the “carla_ Town01” folder to the "modules/map/data" folder?

**Reply by wohu1104** [reply]
the "carla_ Town01" folder should be in the "modules/map/data" folder, we will check the code.

**Reply by Leason-ly** [reply]
THANKS A LOT！！
I’ll really appreciate if you can share the correct version of you apollo source code later. I will try on my own and I hope I can fix the problem. 
your source code committed before maybe has something wrong with the “dreamview” module as it cannot play back the record data
.
All WISHES!

**Reply by synkrotron** [reply]
Hello @Leason-ly , We have updated the guardstrike/apollo_carla_bridge branch and added  missing files carla_Town01. Please have a try. Thanks for your feedback.
